src/nvim/api/window.c

```diff
@@ -366,6 +366,10 @@ Integer nvim_win_get_number(Window window, Error *err)
 Boolean nvim_win_is_valid(Window window)
   FUNC_API_SINCE(1)
 {
+  if (window == kObjectTypeNil){
```

Comment:
I'm not sure if this is the right fix. `Window` (the type) is just an integer, and `kObjectTypeNil` is an enum with value 0. So this would break `nvim_win_is_valid(0)`.

The problem is that `nil` is being implicitly converted to `0` instead of throwing an error. This is happening in the `GENERATE_INDEX_FUNCTION` macro which converts the value to a `Window` type using `lua_tonumber`, but `lua_tonumber` converts `nil` to `0`.